blog/_posts/2015-12-2-includeos-is-now-open-source.md

```diff
@@ -0,0 +1,38 @@
+
+### We're making a github-based package manager
+One very important feature we'll be needing in order to make IncludeOS grow, is a package manager. We're making one much like [npm](https://www.npmjs.com/), where anyone can create a package as a github repository, and then register that package in a central location for others to use. The idea is to have `$ include http` on the terminal, result in the possibility of writing `#include <http>` inside your IncludeOS service. This will obviously take some time, but hopefully we have something working by summer 2016.
+
+## Networking from scratch
+One thing you'll notice is that our network stack is currently being written from scratch. Are we crazy? Possibly. We obviously considered - and wanted to - port the stack from somewhere else, but not being intimately familiar with any existing kernel, I found that to be even harder than just doing it from scratch. I got a lot of help from reading the nicely annotated and cross-linked [SanOS source code](http://www.jbox.dk/sanos/source/), and once the virtio network driver was working, the rest wasn't nearly as hard as one might think - at least up to now. Clearly it will take time to make it complete, but the efforts it took to get form a driver, up to rudimentary TCP was really insignificant compared to the challenges we've faced when trying to integrate third-party libraries such as newlib (Standard C) and libc++ (Standard C++). And - getting to read RFC's and implementing the TCP handshake from scratch has really been a blast - a highly recommended exercise.
+
+
+## C++ is a new language
+If you haven't looked at C++ after 2011, you should look now -  "It feels like a new language" (Bjarne Stroustrup). While you can always argue that adding new feature won't necessarily make the language any better, it most definitely makes it more powerful. Here are a couple of really nice new language features we're using in IncludeOS:
+
+1. Real Lambdas. Not function pointers, not just call operators, but lambdas just like you know and love from functional languages.
+2. C++11 implementation of delegates (i.e. pointers to a member function in a class instance), which are exactly as fast as function pointers. We're using these everywhere. Reading [the implementation](https://github.com/hioa-cs/IncludeOS/blob/master/api/utility/delegate.hpp) is a great way to have your brain explode, but they work like magic.
+4. User defined literals, and std::literals for units. You can set timers by passing in `100ms`, `10s` or `50min` as parameters to the same function. 
+3. A lot of wisdom, packed into concise [guidelines from the Jedi Council](https://github.com/isocpp/CppCoreGuidelines).
```

Comment:
I'm embarrassed to say that I got confused by the reference to the Jedi Council here. :)
Not everyone will understand this so I think it might be better to elide it or put it in parentheses (i.e "guidelines from FOO (aka the Jedi Council)"